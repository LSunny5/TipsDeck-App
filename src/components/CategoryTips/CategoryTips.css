.CategoryTipsBox {
    padding: 2%;
    padding-bottom: 5rem;
}

.CategoryTipsBox > h1 {
    padding: 3%;
}

.TipsBox {
    display: flex;
    flex-direction: column; 
    justify-content: center;
    align-items: center;
    overflow: hidden;
}

.tName {
    font-size: 1.2em;
    padding: 0.5rem;
}

.TipLink {
    padding: 1rem;
    margin: 2%;
    text-decoration: none;
    color: black;
    width: 60%;
    height: 200px;
    border-radius: 30px;
    background: white; 
    border: 3px solid #FF8E29;
    -webkit-transition: all .15s ease-in-out;
    transition: all .15s ease-in-out;
}

.TipLink:hover {
    background: #FFBA66;
    box-shadow: 0 0 10px 0 #FF8E29 inset, 0 0 20px 2px #FF8E29;
    border: 3px solid #FF8E29;
}

.TipLink:active {
    background: #D9A86C;
}

/* tip description css */
.tipDesc {
    max-height: 6rem;
    /* line clamping so ellipsis is used after 3 lines */
    display: -webkit-box;
    -webkit-line-clamp: 5;
    -webkit-box-orient: vertical; 
    overflow: hidden;
    text-overflow: ellipsis;
    font-size: 0.8em;
    word-wrap: break-word;
}

@media only screen and (min-width: 600px) {   
    .TipsBox {
        flex-direction: row;
        flex-wrap: wrap;
    }

    .tName{
        font-size: 1em;
    }

    .TipLink {
        width: 25%;
        margin: 1rem;
        overflow: hidden;
        min-height: 20vh;
    }
}

@media only screen and (min-width: 1024px) { 
    .CategoryTipsBox{
        min-height: 88vh;
        padding: 1rem;
    }

    .CategoryTipsBox > h1 {
        padding: 1rem;
    }

    .tipDesc {
        max-height: 10rem;
        -webkit-line-clamp: 6;
    }

    .TipLink {
        width: 20%;  
        height: 25vh;
    }
}