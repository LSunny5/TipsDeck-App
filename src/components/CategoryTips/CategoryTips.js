import React from 'react';
import './CategoryTips.css';
import TipsDeckContext from '../../TipsDeckContext';
import { getTips, getCategoryId} from '../../TipsDeckHelpers';
import { NavLink } from 'react-router-dom';

//List all the tips from the Category
class CategoryTips extends React.Component {
    static contextType = TipsDeckContext;

    render() {
        const { tips = [], categories = [] } = this.context;
        const { name } = this.props.match.params;
        //category ID to be used to search for the tips in the category
        let cat = getCategoryId(categories, name) || {};
        //get the list of tips for each category
        let categoryTips = getTips(tips, cat.id);

        return (
            <section className="CategoryTipsBox">
                <h1>{name}</h1>

                <div className="TipsBox">
                    {(categoryTips.length > 0) ? (
                        categoryTips.map(tip =>
                            <NavLink
                                key={tip.id}
                                to={`/Category/${name}/${tip.id}`}
                                className="TipLink">
                                <h2 className="tName">{tip.name}</h2>
                                <p className="tipDesc">{tip.description}</p>
                            </NavLink>
                        )) : (
                            <div className="noTips">
                                Sorry there are no tips at this time.
                            </div>)}
                </div>
            </section>
        );
    };
}

export default CategoryTips;