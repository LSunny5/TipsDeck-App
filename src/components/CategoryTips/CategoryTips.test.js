import React from 'react';
import ReactDOM from 'react-dom';
import CategoryTips from './CategoryTips';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`CategoryTips component`, () => {
    const props = {
        match: {
          params: {
            category: 'Laundry'
          }
        }
      }

    it('renders without crashing', () => {
        const div = document.createElement('div');     
        ReactDOM.render(
            <BrowserRouter>
                <CategoryTips {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the CategoryTips component', () => {
        const wrapper = shallow(<CategoryTips {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});