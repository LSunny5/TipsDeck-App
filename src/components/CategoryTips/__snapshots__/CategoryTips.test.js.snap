// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTips component renders the CategoryTips component 1`] = `
<section
  className="CategoryTipsBox"
>
  <h1 />
  <div
    className="TipsBox"
  >
    <div
      className="noTips"
    >
      Sorry there are no tips at this time.
    </div>
  </div>
</section>
`;
